src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1114,113 +1106,192 @@ public String getFileName()
+         * {
+         *     attachment.downloadToFile()
+         *         .thenAccept(file -> System.out.println("Saved attachment to " + file.getName()))
+         *         .exceptionally(t ->
+         *     { // handle failure
+         *         t.printStackTrace();
+         *         return null;
+         *     });
+         * }
+         * }</pre>
          *
-         * @return boolean true, if successful, otherwise false
+         * @return {@link java.util.concurrent.CompletableFuture} - Type: {@link java.io.File}
          */
-        public boolean download(File file)
+        public CompletableFuture<File> downloadToFile() // using relative path
         {
-            try
-            {
-                withInputStream((in) -> Files.copy(in, Paths.get(file.getAbsolutePath())));
-                return true;
-            }
-            catch (Exception e)
-            {
-                JDAImpl.LOG.error("Error while downloading an attachment", e);
-            }
-            return false;
+            return downloadToFile(getFileName());
         }
 
         /**
-         * Creates a copy of the {@link java.io.InputStream InputStream} that is created using an {@link okhttp3.OkHttpClient OkHttpClient}.
-         *
-         * <p>You can access the input stream directly using {@link #withInputStream(net.dv8tion.jda.api.utils.IOConsumer) withInputStream(IOConsumer)}
-         * which will have an open input stream available within the consumer scope. The stream will be closed once that method returns.
+         * Downloads the attachment to a file at the specified path (relative or absolute).
+         * <br>This will download the file using the {@link net.dv8tion.jda.api.JDA#getCallbackPool() callback pool}.
+         * Alternatively you can use {@link #retrieveData()} and use a continuation with a different executor.
          *
-         * @throws java.io.IOException
-         *         If an IO error occurs trying to read from the opened HTTP channel
+         * <h2>Example</h2>
+         * <pre>{@code
+         * public void saveLocally(Message.Attachment attachment)
+         * {
+         *     attachment.downloadToFile("/tmp/" + attachment.getFileName())
+         *         .thenAccept(file -> System.out.println("Saved attachment to " + file.getName()))
+         *         .exceptionally(t ->
+         *     { // handle failure
+         *         t.printStackTrace();
+         *         return null;
+         *     });
+         * }
+         * }</pre>
          *
-         * @return InputStream copy of the response body for this Attachment
+         * @param  path
+         *         The path to save the file to
          *
-         * @since  3.4.0
+         * @return {@link java.util.concurrent.CompletableFuture} - Type: {@link java.io.File}
          */
-        public InputStream getInputStream() throws IOException
+        public CompletableFuture<File> downloadToFile(String path)
```

Comment:
We should have an overload of `downloadToFile(File file)`

Reply:
Should it replace this overload?

Reply:
No. I would have one feed into the other.